Shock/Classes/NIOHTTPHander.swift

```diff
@@ -0,0 +1,132 @@
+        switch method {
+        case .HEAD:
+            return "HEAD"
+        case .GET:
+            return "GET"
+        case .POST:
+            return "POST"
+        case .PUT:
+            return "PUT"
+        case .PATCH:
+            return "PATCH"
+        case .DELETE:
+            return "DELETE"
+        default:
+            return ""
+        }
+    }
+    
+    private func requestForHTTPRequestHead(_ request: HTTPRequestHead) -> NIOHTTPRequest? {
+        guard let url = URLComponents(string: request.uri) else { return nil }
+        let path = url.path
+        let method = stringForHTTPMethod(request.method)
+        let headers = request.headers.reduce(into: [String: String](), { $0[$1.0.lowercased()] = $1.1 })
+        let body = [UInt8]()
+        let address = url.host
+        var params = [String: String]()
+        var queryParams =  [(String, String)]()
+        if let queryItems = url.queryItems {
+            params = queryItems.reduce(into: [String: String](), { $0[$1.name] = $1.value })
+            queryParams = queryItems.reduce(into: [(String, String)](), { $0.append(($1.name, $1.value ?? "")) })
+        }
+        
+        return NIOHTTPRequest(path: path,
+                              queryParams: queryParams,
+                              method: method,
+                              headers: headers,
+                              body: body,
+                              address: address,
+                              params: params)
+    }
+    
+    private func handleResponse(_ response: HttpResponse, for request: HTTPRequestHead, in context: ChannelHandlerContext) {
+        switch response {
+        case .raw(_, _, let customHeaders, let handler):
+            if let handler = handler {
+                let writer = NIOHTTPResponseBodyWriter()
+                do {
+                    try handler(writer)
+                    var headers = HTTPHeaders()
+                    headers.add(name: "content-length", value: "\(writer.contentLength)")
+                    if let customHeaders = customHeaders {
+                        for (name, value) in customHeaders {
+                            headers.add(name: name, value: value)
+                        }
+                    }
+                    _ = context.write(self.wrapOutboundOut(.head(httpResponseHeadForRequestHead(request, status: .ok, headers: headers))))
+                    if writer.contentLength > 0 {
+                        _ = context.writeAndFlush(self.wrapOutboundOut(.body(.byteBuffer(writer.buffer))), promise: nil)
+                    }
+                } catch {
+                    _ = context.writeAndFlush(self.wrapOutboundOut(.head(httpResponseHeadForRequestHead(request, status: .internalServerError))))
+                }
+            } else {
+                _ = context.writeAndFlush(self.wrapOutboundOut(.head(httpResponseHeadForRequestHead(request, status: .internalServerError))))
```

Comment:
could this be turned into a local helper method? It's very verbose and only seems to have two possible differences (the status and optional headers).